### PR TITLE
feat: allow browserlist configs more easily

### PR DESCRIPTION
**Pull request by jquense**
The overall intent is to allow our internal packages to extend or specify a browserlistrc, which is more portable and also works with autoprefixer.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -35,27 +37,46 @@ const defaultOptions = {
   runtime: false,
   corejs: false,
   debug: false,
-  ignoreBrowserslistConfig: true,
+  ignoreBrowserslistConfig: false,
   browsers: defaultBrowsers,
+  include: [
+    // XXX: Workaround for babel/babel#8019.
+    'transform-classes',
```
Comment (jquense) [inline index.js]: felt like it made sense to move these here.

File: package.json [hunk 2]
```diff
@@ -39,6 +39,7 @@
     "testEnvironment": "node"
   },
   "devDependencies": {
+    "@4c/babel-preset-4catalyzer": "file:../babel-preset-4catalyzer",
```
Comment (jquense) [inline package.json]: lol

File: index.js [hunk 3]
```diff
@@ -35,27 +37,46 @@ const defaultOptions = {
...
 };
 
+function getTargets(
+  env,
+  { target, targets, configPath, ignoreBrowserslistConfig },
+) {
+  if (env !== 'production') {
+    return target === 'node' ? { node: 'current' } : { esmodules: true };
+  }
+  if (target === 'node') return { node: '10.0' };
```
Comment (taion) [inline index.js]: bleh we should distinguish "app" build targets from "lib" ones, but that can be later

Comment (taion) [reply index.js]: though a TODO would be appropriate

**Pull request merged**